### PR TITLE
Example on how to automatically refresh the browser when Blazor Server reconnection fails

### PR DESCRIPTION
**Pull request by MackinnonBuck**
## Example on how to automatically refresh the browser when Blazor Server reconnection fails

This PR adds an example showing how to configure a Blazor Server app to automatically refresh the browser if reconnection to the server fails.

### Description

Customers have been asking for a way to keep the existing Blazor Server reconnection logic and UI, but modified slightly so that refreshing the browser happens automatically rather than requiring direct user action. We don't have an extensibility point in our current JavaScript API that allows this do be done without reimplementing the full reconnection logic and UI. That said, it's not _that_ much code, so it could be sufficient to provide an example in the docs showing how to achieve this.

I'm opening this initially as a draft PR because there are a few open questions:
1. How complete should the example code be?
    * Should we make the example more complete by including some example CSS for the `reconnect-modal`?
    * Or, should we take the opposite approach and reduce the example by simplifying the reconnect functionality?
      * I'm personally not in favor of this, because the customer ask is to retain existing reconnect functionality but with automatic browser refreshing.
2. Is pasting the code inline the best approach?
    * While the code is straightforward, it's a bit on the longer end. Should we put this code elsewhere and link to it from the docs?
3. Rather than having a docs example for this at all, should we address https://github.com/dotnet/aspnetcore/issues/32113 and add new APIs to enable this scenario without requiring reconnection logic to be reimplemented?

Fixes https://github.com/dotnet/aspnetcore/issues/44715

### CODE REVIEW

**Review by danroth27 — APPROVED** [review]
This seems fine to me, and I don't think the code is too long to include inline.

**Comment by danroth27** [comment]
@MackinnonBuck It looks like https://github.com/dotnet/aspnetcore/issues/32113 may be a duplicate of https://github.com/dotnet/aspnetcore/issues/44669, which is already being considered for .NET 8. Thoughts?

File: aspnetcore/blazor/fundamentals/signalr.md [hunk 1]
```diff
@@ -326,6 +326,86 @@ To modify the connection events, register callbacks for the following connection
 </body>
 ```
 
+### Advanced scenario: Automatically refresh the page when reconnection fails
```
Comment (guardrex) [inline aspnetcore/blazor/fundamentals/signalr.md:329]: We almost never tag headings this way. Can it just call out what this content covers? ...

```suggestion
### Automatically refresh the page when reconnection fails
```

File: aspnetcore/blazor/fundamentals/signalr.md [hunk 2]
```diff
@@ -326,6 +326,86 @@ To modify the connection events, register callbacks for the following connection
...
+
+    <div id="reconnect-modal" style="display: none;"></div>
+    <script src="_framework/blazor.server.js" autostart="false"></script>
+    <script src="boot.js"></script>
+</body>
+```
+
+`wwwroot/boot.js`:
+
+```js
```
Comment (guardrex) [inline aspnetcore/blazor/fundamentals/signalr.md:347]:
```suggestion
```javascript
```

**Review by guardrex — APPROVED** [review]
LGTM with three suggestions.

File: aspnetcore/blazor/fundamentals/signalr.md [hunk 3]
```diff
@@ -326,6 +326,86 @@ To modify the connection events, register callbacks for the following connection
 </body>
 ```
 
+### Advanced scenario: Automatically refresh the page when reconnection fails
+
```
Comment (guardrex) [inline aspnetcore/blazor/fundamentals/signalr.md:330]:
```suggestion

*This section applies to Blazor Server apps.*
```

**Comment by guardrex** [comment]
> Should we make the example more complete by including some example CSS for the reconnect-modal?

Your call; but if you do include it and want to cross-link, we cover that at ...

\[Reflect the connection state in the UI (Blazor Server)](#reflect-the-connection-state-in-the-ui-blazor-server)

**Comment by MackinnonBuck** [comment]
> @MackinnonBuck It looks like [dotnet/aspnetcore#32113](https://github.com/dotnet/aspnetcore/issues/32113) may be a duplicate of [dotnet/aspnetcore#44669](https://github.com/dotnet/aspnetcore/issues/44669), which is already being considered for .NET 8. Thoughts?

@danroth27 I think those seem like two different issues.

From my understanding, the former is asking for an extra `onReconnectionFailed` event on the default reconnection handler so you can add the automatic refreshing behavior in just a few lines of code.

The latter seems to be asking for a way for `Blazor.reconnect()` to signal whether reconnection failed due to the circuit ID being invalid or due to something like the server being down. Addressing this would still require that customers provide a full implementation of the reconnection logic in order to automatically refresh the browser on reconnection failure.

But now that you called out https://github.com/dotnet/aspnetcore/issues/44669, I think this is something we already have support for, and it's what my sample in this PR utilizes. Our own [default reconnection handler](https://github.com/dotnet/aspnetcore/blob/921f5417741ca8086bd67d437ad40a91575095a1/src/Components/Web.JS/src/Platform/Circuits/DefaultReconnectionHandler.ts#L78-L94) even handles both cases explicitly ("failed" vs "rejected"). Maybe that issue can be closed? CC @SteveSandersonMS

**Comment by guardrex** [comment]
@MackinnonBuck ... You might be waiting on Steve to look. I'll let you pull the trigger on merging this when ur ready.

**Pull request merged**